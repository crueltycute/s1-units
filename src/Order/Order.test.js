import { shallow, configure } from 'enzyme';
import React from 'react';

import { fakeOrders } from '../mock/fakeOrders';
import Order from './Order';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('Order component', () => {
    const wrapper = shallow(<Order order={fakeOrders[0]} />);
    const wrapperSecond = shallow(<Order order={fakeOrders[1]} />);
    const wrapperNotValid = shallow(<Order order={{}} />);

    it('render first element', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render second element', () => {
        expect(toJson(wrapperSecond)).toMatchSnapshot();
    });

    it('render empty element', () => {
        expect(toJson(wrapperNotValid)).toMatchSnapshot();

        expect(toJson(wrapperNotValid)).not.toEqual(toJson(wrapper))
    });
})