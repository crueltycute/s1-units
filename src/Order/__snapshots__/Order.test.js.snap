// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component render empty element 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    />
    <span>
      NaN undefined, undefined, NaN год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order component render first element 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Ali Express
    </span>
    <span>
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order component render second element 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Lamoda.ru
    </span>
    <span>
      13 марта, ср, 2019 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Жакет - BOREAL5
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Miss Gabby Костюм
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Ostin перчатки мужские
    </div>
    <div
      className="Order-item"
      key="3"
    >
      Zara худи роз.
    </div>
  </div>
</div>
`;
